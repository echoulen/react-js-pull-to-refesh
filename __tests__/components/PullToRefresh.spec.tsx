import * as React from "react";
import * as renderer from "react-test-renderer";
import {PullToRefresh} from "../../src/";

describe("PullToRefresh spec", () => {
    it("App shows PullToRefresh", () => {
        const element = renderer
            .create((
                <PullToRefresh
                    onRefresh={null}
                    pullDownContent={<div>Pull Down</div>}
                    releaseContent={<div>Release to refresh</div>}
                    refreshContent={<div>Refreshing</div>}
                    pullDownThreshold={200}
                    triggerHeight={100}
                    backgroundColor="white"
                    startInvisible={false}
                >
                    <div>Test</div>
                </PullToRefresh>
            ))
                .toJSON();
        expect(element).toMatchSnapshot();
    });
});
