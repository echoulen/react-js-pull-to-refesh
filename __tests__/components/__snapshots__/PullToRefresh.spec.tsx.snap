// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PullToRefresh spec App shows PullToRefresh 1`] = `
<div
  style={
    Object {
      "WebkitOverflowScrolling": "touch",
      "backgroundColor": "white",
      "height": "auto",
      "overflow": "hidden",
      "position": "relative",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "left": 0,
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "visibility": "visible",
      }
    }
  >
    <div>
      Pull Down
    </div>
  </div>
  <div
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "backgroundColor": "white",
        "height": "auto",
        "overflow": "hidden",
        "position": "relative",
        "zIndex": 1,
      }
    }
  >
    <div>
      Test
    </div>
  </div>
</div>
`;
